Bind auth api

thanghq-zinza opened the pull request.

thanghq-zinza committed:
feat: bind login, forgot pwd, reset pwd api

thanghq-zinza committed:
feat: bind sign up api

hungnp194 commented on src/app/(authPage)/register/page.tsx:
```diff
@@ -84,36 +108,45 @@ const Register: FC = () => {
-    router.push('/');
+    console.log(formData);
+
+    await fetchAPI('auth/sign-up', 'POST', formData)
+      .then(() => {
+        toast.success('Đăng ký thành công');
+        router.push('/login');
+      })
+      .catch((error) => {
+        console.log(error);
+        toast.error('Đăng ký thất bại');
+        setLoading(false);
+      });
   };
```
api viết vào trong custom hook, viết type cho response, request data

hungnp194 commented on src/app/(authPage)/reset-password/[token]/page.tsx:
```diff
@@ -0,0 +1,127 @@
+    router.push('/login');
+  };
+
+  const obSubmit = async (data: FormData) => {
```
typo

hungnp194 commented on src/app/(authPage)/reset-password/[token]/page.tsx:
```diff
@@ -0,0 +1,127 @@
+  const obSubmit = async (data: FormData) => {
+    setLoading(true);
+
+    await fetchAPI(`auth/change-password/${params.token}`, 'POST', {
+      password: data.password
+    })
+      .then((response) => {
+        toast.success('Đổi mật khẩu thành công!');
+        goToLogin();
+      })
+      .catch((error) => {
+        console.log(error);
+        toast.error('Có lỗi xảy ra hoặc yêu cầu đổi mật khẩu đã hết hạn!');
+      });
+
+    setLoading(false);
```
try catch

thanghq-zinza committed:
fix: add auth state to user slice

thanghq-zinza committed:
fix: fix api calls, add authentication rule

Use custom hook with react query and axios for api calls
Add authentication rule for pages
Fix typo

thanghq-zinza committed:
fix: add response type to apis

thanghq-zinza committed:
fix: fix api hooks
